### PR TITLE
feat: add automatic git remote configuration

### PR DESCRIPTION
**Pull request by gfranxman**
Added functionality to automatically configure git remote repositories to the scaf creation process.

This is done in the post_gen_project.py hook file using a new function called `configure_git_remote()`.  This function takes the repository URL built during the cookiecutter build out and initializes a new git repository in the new project directory.  It then sets the entered URL as the 'origin' remote.

Subsequently, removed the `git init .` statement from the scaf script since it now occurs in the `configure_git_remote()` function of the post_gen_project hook.

Also provided a detailed guide in README.md on how to test the cookiecutter portion of scaf.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,3 +109,66 @@ When making changes to scaf, keep the following in mind:
...
+OUTPUT_FOLDER=./scaf-test
+TEST_CONFIG=./scaf-cookiecutter.yaml
+
+BRANCH_NAME=gfranxman/configure-git-remote
+BASE_REPO=https://github.com/sixfeetup/scaf.git
+
+
+rm -rf $OUTPUT_FOLDER && \
+$SCAF_ROOT/scaf myproject --no-input --checkout $BRANCH_NAME ${BASE_REPO} 
+```
```
Comment (rochecompaan) [inline README.md:174]: This deserves a separate issue and PR where we work out the testing details in a more reusable way. For instance, we can have Makefile  target called `test` and `test-remote`:
```
BRANCH_NAME = $(shell git rev-parse --abbrev-ref HEAD)
BASE_REPO = https://github.com/sixfeetup/scaf.git
SCAF_ROOT = $(shell pwd)

test:
	echo ${BRANCH_NAME}
	rm -rf /tmp/scaf-test && \
	mkdir /tmp/scaf-test && \
	./scaf myproject -o /tmp/scaf-test --no-input .     # this does not currently work, but it should

test-remote:
	echo ${BRANCH_NAME}
	rm -rf /tmp/scaf-test && \
	mkdir /tmp/scaf-test && \
	cd /tmp/scaf-test && \
	cp ${SCAF_ROOT}/cookiecutter.json . && \
	${SCAF_ROOT}/scaf myproject \
		--no-input \
		--checkout ${BRANCH_NAME} ${BASE_REPO}
```

File: README.md [hunk 2]
```diff
@@ -109,3 +109,66 @@ When making changes to scaf, keep the following in mind:
 - update pins in requirements/*.in files but *don't\* commit the compiled requirements.txt
   files to the repo.
 - update to the latest Python supported by Django. For Django 4.1, this is 3.8, 3.9, and 3.10.
+
+### Testing
+To test the cookiecutter portion of scaf, create virtual environment ad install black, isort and cookiecutter.
+Create a test cookiecutter.yaml file with the following content:
```
Comment (rochecompaan) [inline README.md:115]: We could just use the default `cookiecutter.json`, see my comment below.

**Review by rochecompaan — CHANGES_REQUESTED** [review]
Looks good!

Can we make the testing more reusable, please? Either in this PR or another.

**Comment by gfranxman** [comment]
Made a ticket for turning the testing docs into code and makefile targets

**Pull request merged**